esm-tools/esm_tools
# hotfix/awiesm compile on mistral

# Hotfix: `AWIESM-2.1` Compiles on Mistral

This merge request allows AWIESM to be compiled on MISTRAL. This worked previously, but somehow broke in the meantime. It *might* require changes to `mistral.yaml`, thus I want to put this online to inform other developers (@seb-wahl for GEOMAR models, @denizural for ICON, @mandresm for MPIESM)

Once we have figured out what exactly needs to change to get AWIESM running again, and if that change affects mistral.yaml, I'd ask you to please test your models before we merge this.....

## To be tested (Compile):

**AWIESM + Standalone Version**
+ [x] AWI-ESM-2.1 on Ollie
+ [x] ECHAM-6.3.05-concurrent-rad on Ollie
+ [x] AWI-ESM-2.1 on Mistral
+ [x] ECHAM-6.3.05-concurrent-rad on Mistral

**AWICM (Standard Version)**
+ [x] AWICM-2.0 on Mistral

**MPIESM**
+ [x] MPIESM compiles in mistral in all its versions

**ICON**
@denizural : Please add whatever you want to test

**GEOMAR Models**
@seb-wahl: Please add whatever you want to test

## How to test:
```
cp ${HOME}/.esmtoolsrc ${HOME}/.esmtoolsrc_backup
cd /wherever/you/like
python -m venv .venv
source .venv/bin/activate
git clone https://github.com/esm-tools/esm_tools.git
cd esm_tools
git checkout hotfix/dkrz_awiesm_2_1
./install.sh
esm_versions check
echo "Please proceed with your tests"
echo "Do not forget to restore the esmtoolsrc backup!!"
```
